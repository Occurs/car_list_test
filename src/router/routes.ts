const routes = {
  carsList: '/',
  carCard: '/car/:stockNumber',
  favorites: '/favorites'
};

export default routes;