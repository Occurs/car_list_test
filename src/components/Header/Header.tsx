import React from "react";
import { NavLink } from "react-router-dom";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import routes from "router/routes";
import { colors } from "styles/variables";

const Header = () => {
  return (
    <Box
      component="header"
      height="80px"
      borderBottom={`2px solid ${colors.secondary}`}
    >
      <Container maxWidth="md">
        <Grid
          container
          direction="row"
          justify="space-between"
          alignItems="center"
        >
          <Grid item>
            <Box component="div">
              <Typography>Logo</Typography>
            </Box>
          </Grid>
          <Grid item>
            <Typography>
              <Box
                component="span"
                height="80px"
                display="flex"
                alignItems="center"
                justifyContent="center"
                fontSize="14px"
              >
                <Link underline="none" component={NavLink} to={routes.carsList}>
                  <Box color={colors.font} component="span" marginLeft="24px">
                    Purchase
                  </Box>
                </Link>
                <Link
                  underline="none"
                  component={NavLink}
                  to={routes.favorites}
                >
                  <Box color={colors.font} component="span" marginLeft="24px">
                    My Orders
                  </Box>
                </Link>
                <Link underline="none" component={NavLink} to="/sell">
                  <Box color={colors.font} component="span" marginLeft="24px">
                    Sell
                  </Box>
                </Link>
              </Box>
            </Typography>
          </Grid>
        </Grid>
      </Container>
    </Box>
  );
};

export default Header;
