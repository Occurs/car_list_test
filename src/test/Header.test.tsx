import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import Header from "components/Header/Header";
import routes from "router/routes";

describe("Header Component", () => {
  test("snapshot", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[routes.carsList]}>
        <Header />
      </MemoryRouter>
      );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
