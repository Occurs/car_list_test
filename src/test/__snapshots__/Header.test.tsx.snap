// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot 1`] = `
<header
  className="MuiBox-root MuiBox-root-1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Logo
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="MuiBox-root MuiBox-root-3"
          >
            <a
              aria-current="page"
              className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary active"
              href="/"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              style={Object {}}
            >
              <span
                className="MuiBox-root MuiBox-root-4"
              >
                Purchase
              </span>
            </a>
            <a
              aria-current={null}
              className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
              href="/favorites"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <span
                className="MuiBox-root MuiBox-root-5"
              >
                My Orders
              </span>
            </a>
            <a
              aria-current={null}
              className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
              href="/sell"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <span
                className="MuiBox-root MuiBox-root-6"
              >
                Sell
              </span>
            </a>
          </span>
        </p>
      </div>
    </div>
  </div>
</header>
`;
